### DISCUSSION TITLE
Include gitleaks in gitlab-server pre commit hook mode?

### DISCUSSION THREAD

**Discussion by Stamka** [zricethezav/gitleaks#1104] [Q&A]
Hi.
I need to run gitleaks in gitlab when developer try to commit his code.
Is there some instructions how can I do it?

**Comment by maltemorgenstern** [comment]
Hey @Stamka,
there are different options available - and the choice depends on when you want to run `gitleaks`.

_Option A:_
If you want to run `gitleaks` locally (e.g. before the commit gets pushed to the GitLab server) then take a look at the `pre-commit` configuration: https://github.com/gitleaks/gitleaks#pre-commit

_Option B:_
If you want to run `gitleaks` on the GitLab server (which would require a self-hosted instance I think) then you should be able to configure a `pre-receive` hook: https://docs.gitlab.com/ee/administration/server_hooks.html

_Option C:_
If you want to run `gitleaks` on the GitLab server as part of a [CI/CD pipeline](https://docs.gitlab.com/ee/ci/) you can use the gitleaks docker image and configure the job to fail if any secrets are detected.

Regards
Malte
